lib/wasi/src/runtime/ws.rs

```diff
@@ -0,0 +1,24 @@
+#[cfg(feature = "async_ws")]
+use async_trait::async_trait;
+
+use crate::WasiRuntimeImplementation;
+
+// This ABI implements a general purpose web socket
+#[cfg_attr(feature = "async_ws", async_trait)]
+pub trait WebSocketAbi {
```

Comment:
I have no idea on what this accomplishes

Reply:
Allows environments to provide different Websocket client implementations.

I already commented previously that I think this should be integrated with my added `wasix_http_client`, but otherwise not be so deeply integrated into the whole code base.

Reply:
Websocket support was removed.